### PR TITLE
1634 Make getFeatureFlagValue return strong type

### PR DESCRIPTION
**Pull request by leite08**
Ref. metriport/metriport-internal#1634

### Dependencies

none

### Description

Make `getFeatureFlagValue` return strong type.

Previously

image

image

With this PR

image

image

### Testing

- Local
  - [x] Retrieves string FF
  - [x] Retrieves boolean FF
- Staging
  - [ ] Update a patient and it runs w/o error
- Sandbox
  - none
- Production
  - none

### Release Plan

- [ ] Merge this

### CODE REVIEW

**Review by RamilGaripov — APPROVED** [review]

File: packages/core/src/external/aws/app-config.ts [hunk 1]
```diff
@@ -20,22 +20,28 @@ export type StringValuesFF = z.infer<typeof ffStringValuesSchema>;
...
 export type BooleanFF = z.infer<typeof ffBooleanSchema>;
 
-export const ffDatastoreSchema = z.object({
+export const booleanFFsSchema = z.object({
+  commonwellFeatureFlag: ffBooleanSchema,
+  carequalityFeatureFlag: ffBooleanSchema,
+});
+export type BooleanFeatureFlags = z.infer<typeof booleanFFsSchema>;
+
+export const stringValueFFsSchema = z.object({
```
Comment (jonahkaye) [inline packages/core/src/external/aws/app-config.ts:32]: I know this is from the previous FF PR, but I was a little confused since StringValuesFF is really BooleanFF + values I had to do some digging to make sure the StringValues FF also included the enabled boolean since it was clear from the name. Could we change the stringValueFFsSchema to stringValueAndBooleanFFsSchema or is that too wordy?

Comment (leite08) [reply packages/core/src/external/aws/app-config.ts:32]: I think its not necessary. If we had 3 or 4 properties we wouldn't add all them on the name 😅
